### ISSUE TITLE
[Card] Double tab stop on clickable cards (i.e., isClickable=true) wrapped in a `Hyperlink`.

### ISSUE THREAD

**Issue by adamstankiewicz** [openedx/paragon#2264]
## Bugs

The Paragon documentation site currently recommends in the "As link" section to wrap the `Card` (with `isClickable`) with a `Hyperlink` in order to render it as a link. This results in a double tab stop, an a11y issue that requires one additional keystroke for screen reader users when navigating through a page (e.g., a grid of 24 cards in search results x 2 tabs each).

The following [Paragon Playground example](https://paragon-openedx.netlify.app/playground/?code=N4Igxg9gJgpiBcJgApkEoAEBeAfB4AOgHYYaREDOALhgJYUCiAHlQE4CGAygLbsA2fbBgCuFGAFkYUWuwCKwmKwCeyYBl5MA6rShUAFvAwAjVjHYBrAA4RaRKhQB0MFhx78%2BDjdt16MAXzQAbmJiUlMqYVYSZFDSDAAeAGF2VigMdgosAhB2bIwqFIBzGCoskAB9Iz52InM86iU%2BGCxgNQB3HX1DemY2Ll4BDAB%2BDGyARgAGCYBSPMNxgA5Tbjy-PzoKRL5aMHN2KpgcWLiE5NSHAEleYuTLY5OMClYwMr0qKksKeAB6b8sdijCbgOSx6CBUCAUb4AZgAbBNvgAmKbfbL3E5PMAAQT4pWyZzStGuMDRJAe3yOZJOSRSUAcAAkzLBWPlaFQmmUCRgACpsprZCno060hycGBgKi0CBESkPOLcvT0DbpMi0x7iyXShwYC40MA1MjSgq2dJEJT6WyFYzCGiiYTuJT5ZxUAA0Ku21DdEBZHpoEAAZhhtrVHKMqQ9xMJcbRLE11RKpZQMHp2AA3GAq-WpDDSVM6RTGEptGAwEj6GDAoXxb4E0UaxOyh4084AMQg4MUjbl8QAQjaITKsQnpRgxtW%2B%2B9pV24tXa22O6xp7PaY2gsQAug-CAXSBy9wYBQEABtAC6fiAA) shows how I think we should recommend using `Card` as a hyperlink in the documentation site to avoid the double tab stop moving forward.

The `react-bootstrap/Card` component our `Card` is based on supports an `as` prop that allows the default `div` to be changed to any HTML element or React component, and allows its props to be passed through.

That said, as you can see in the Playground, the text gets the hyperlink blue and the text decoration on hover, which is not intended, as it should be relying on the hover/focus styles for the entire `Card` itself.

```[tasklist]
### Tasks
- [ ] Update docs site example to change the "As link" example to `<Card as={Hyperlink} destination={...}>` to resolve the double tab stop a11y issue.
- [ ] Resolve styling when `as={Hyperlink}` or `as="a"` or `as={Link}` is passed to `Card` component.
```
